gems/aws-sdk-core/lib/aws-sdk-core/rest/response/body.rb

```diff
@@ -30,6 +31,13 @@ def apply(body, data)
 
         private
 
+        def empty_blob_body?(body)
+          return false if !@rules[:payload_member].shape.is_a?(BlobShape) ||
+                          @rules[:payload_member]['streaming']
+
+          true if body.respond_to?(:size) && body.size.zero?
```

Comment:
These operands already return a boolean, no need to return true or false accordingly, you can structure this method to just return the output of whether the shape is a Blob, body size 0, etc. Could also break this into two methods like "is_streaming?"